src/lib.rs

```diff
@@ -1,6 +1,147 @@
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
+extern crate memmap;
+extern crate byteorder;
+extern crate time;
+#[macro_use] extern crate bitflags;
+#[cfg(unix)] extern crate nix;
+#[cfg(windows)] extern crate kernel32;
+#[cfg(test)] extern crate rand;
+
+use byteorder::{LittleEndian, WriteBytesExt};
+use memmap::{Mmap, Protection};
+use std::ffi::CString;
+use std::fs::OpenOptions;
+use std::io::Cursor;
+use std::io::prelude::*;
+
+const HDR_LEN: usize = 40;
+const TOC_BLOCK_LEN: usize = 16;
+
+type Endian = LittleEndian;
+
+#[cfg(unix)]
+fn get_process_id() -> i32 {
+    nix::unistd::getpid()
+}
+
+#[cfg(windows)]
+fn get_process_id() -> i32 {
+    unsafe { kernel32::GetCurrentProcessId() as i32 }
+}
+ 
+bitflags! {
+    pub struct MMVFlags: u32 {
+        const NOPREFIX = 1;
+        const PROCESS  = 2;
+        const SENTINEL = 4;
+    }
+}
+
+pub struct MMV {
```

Comment:
I called this `Client` in my implementation. We can probably decide a better name than MMV.